scipy/_lib/_util.py

```diff
@@ -647,3 +647,53 @@ def _nan_allsame(a, axis, keepdims=False):
             return np.full(shp, fill_value=True, dtype=bool)
     a0 = _first_nonnan(a, axis=axis)
     return ((a0 == a) | np.isnan(a)).all(axis=axis, keepdims=keepdims)
+
+
+def _rename_parameter(old_name, new_name, dep_version=None):
+    """
+    Generate decorator for backward-compatible keyword renaming.
+
+    Apply the decorator generated by `_rename_parameter` to functions with a
+    recently renamed parameter to maintain backward-compatibility.
+
+    After decoration, the function behaves as follows:
+    If only the new parameter is passed into the function, behave as usual.
+    If only the old parameter is passed into the function (as a keyword), raise
+    a DeprecationWarning if `dep_version` is provided, and behave as usual
+    otherwise.
+    If both old and new parameters are passed into the function, raise a
+    DeprecationWarning if `dep_version` is provided, and raise the appropriate
+    TypeError (function got multiple values for argument).
```

Comment:
```suggestion
    If both old and new parameters are passed into the function, raise a
    TypeError (function got multiple values for argument).
```

Looking at the code, this seems to be the behavior when both parameters are passed.

Reply:
Fixed this. I wanted both to happen.